### DISCUSSION TITLE
How JNode works and how to create OSes in Java like that?

### DISCUSSION THREAD

**Discussion by Ambijubi** [jnode-revisited/jnode#19] [Q&A]
Maybe this is an offtopic, but I VERY want to know it.
I want to create my own OS like MS-DOS, that brings a little nostalgia, but have some modern features in Java (I know Java and learning C++).
But I can't find any good tutorials how to make operating systems in Java, and probably most accessible operating system in Java that i found is JNode.
This is very good OS, but I didn't find explanations how it works.
Tell me, please, How it works and how to create operating systems in Java?

**Comment by tripleo1** [comment]
Do some research and document what you find, knowing that much of it will be either wrong or misinformed.  

You are welcome to ask more specific questions here -- there is not a lot of work on OSes written in Java, and not much written about JNode.

> This is very good OS

This is an interesting statement.

**Reply by tripleo1** [reply]
https://www.youtube.com/watch?v=9GDX-IyZ_C8

**Reply by tripleo1** [reply]
Learn Rust and read this:
https://github.com/phil-opp

Consider sponsorship, too.

Good luck.

**Comment by tripleo1** [comment]
1) what is a jubi?
2) that other thing is a virus.
